component/loki/process/internal/stages/metric.go

```diff
@@ -108,6 +117,17 @@ func newMetricStage(logger log.Logger, config MetricsConfig, registry prometheus
 	}), nil
 }
 
+func registerOrReuse(registerer prometheus.Registerer, c prometheus.Collector) (prometheus.Collector, error) {
+	err := registerer.Register(c)
+	if err == nil {
+		return c, nil
+	}
+	if previous, ok := err.(prometheus.AlreadyRegisteredError); ok {
+		return previous.ExistingCollector, nil
+	}
+	return nil, err
```

Comment:
Should this use [errors.As](https://pkg.go.dev/errors#As) to future proof against wrapped error values?

Reply:
I made this change, then thought that I should test this and when testing I've found that it's safe to MustRegister, because we are using unchecked metrics: https://github.com/grafana/agent/blob/c6b4c3749863553907ee4fb24bc7263fe3efe7c4/component/loki/process/internal/metric/metricvec.go#L36 

It confused me, so I've added comments instead of my change. Not sure I like that checked/unchecked metrics have the same type, but fundamentally different behaviour, but it's a different problem.